octomap/include/octomap/OcTreeBaseImpl.h

```diff
@@ -108,6 +111,57 @@ namespace octomap {
     inline unsigned int getTreeDepth () const { return tree_depth; }
 
     inline double getNodeSize(unsigned depth) const {assert(depth <= tree_depth); return sizeLookupTable[depth];}
+    
+    
+    // -- Tree structure operations formerly contained in the nodes ---
+   
+    /// Creates (allocates) the i-th child of the node. @return ptr to newly create NODE
+    NODE* createNodeChild(NODE* node, unsigned int childIdx);
```

Comment:
Good one! That's almost a philosophical question I'd say. The underlying tree structure changes, although no direct member is touched. The same could be said about `deleteNodeChild(...)`, but what keeps me from calling them on the root node? Then there might be a real effect.

If I'm reading it right, it's exactly what is disussed here: https://isocpp.org/wiki/faq/const-correctness#logical-vs-physical-const